### PR TITLE
testlibusb example ported from libusb-0.1 to libusb-1.0

### PR DESCRIPTION
**Pull request by anilgit90**
Based on Nathans branch https://github.com/hjelmn/libusb-darwin/blob/master/examples/testlibusb1.c.
Added few modification to functions print_bos() and print_endpoint().

### CODE REVIEW

**Comment by anilgit90** [comment]
I suppose my pull request has conflict with the master branch let me resolve it.

**Comment by anilgit90** [comment]
I suppose this is a proper way, Please let me know if otherwise.

File: appveyor.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ build_script:
     msbuild %libusb_2012% /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"
 
     msvc/appveyor.bat
-
+
```
Comment (LudovicRousseau) [inline appveyor.yml]: Why do you add spaces here?

I guess your changes in appveyor.yml are not needed and made by mistake.

**Comment by LudovicRousseau** [comment]
1. ``git rebase`` your changes over ``libusb/master``
2. merge your patches into only 1 since they only concern the same change

The goal is to have only 1 commit in your Pull Request.

**Comment by anilgit90** [comment]
This is okay I guess. Sorry for the troubles. :)

**Comment by anilgit90** [comment]
Hi, Could you let me know what is the issue with this commit. Let me know if it requires any rectification.

**Comment by hjelmn** [comment]
libusb developers are just a little short on time. One of us will get to this soon.

**Pull request closed without merging**